src/bot.ts

```diff
@@ -109,9 +118,29 @@ export default class {
 
     const spinner = ora('📡 connecting to Discord...').start();
 
-    this.client.once('ready', () => {
+    this.client.once('ready', async () => {
       debug(generateDependencyReport());
 
+      spinner.text = '📡 Updating commands in all guilds...';
+
+      // Update commands
+      const rest = new REST({version: '9'}).setToken(this.token);
+
+      if (this.isProduction) {
+        await rest.put(
+          Routes.applicationCommands(this.client.user!.id),
+          {body: this.commandsByName.map(command => command.slashCommand ? command.slashCommand.toJSON() : null)},
+        );
+      } else {
+        // If development, set commands guild-wide
+        this.client.guilds.cache.each(async guild => {
+          await rest.put(
```

Comment:
I think you probably want to wrap this in a `Promise.all(...)` (using `.cache.map` instead of `.cache.each`) so we wait for all the API calls to succeed. I don't think it's currently waiting for Promises to resolve.

Reply:
Done, check if it is correct

Reply:
Promises are now native so you don't need bluebird unless you're doing something fancy. :)

Updated.